src/components/Profile/components/OfficeInfoList/UpdateOfficeHoursDialog/index.jsx

```diff
@@ -0,0 +1,63 @@
+import { FormControl, IconButton, Input, InputLabel, Select, MenuItem } from '@mui/material';
+import EditIcon from '@mui/icons-material/Edit';
+import GordonDialogBox from 'components/GordonDialogBox';
+import GordonSnackbar from 'components/Snackbar';
+import { useState } from 'react';
+import userService from 'services/user';
+
+const UpdateOfficeHours = () => {
+  const [open, setOpen] = useState(false);
+  const [hours, setHours] = useState('');
+  const [snackbar, setSnackbar] = useState({ message: '', severity: null, open: false });
+
+  const handleSubmit = async () => {
+    try {
+      await userService.updateOfficeHours(hours);
+      createSnackbar('Your office hours will update within a couple hours.', 'success');
+    } catch {
+      createSnackbar('Office hours failed to update. Please contact CTS.', 'error');
+    }
+    setOpen(false);
+  };
+
+  const createSnackbar = (message, severity) => {
+    setSnackbar({ message, severity, open: true });
```

Comment:
feel free to push back on me here: is there any real benefit to having a helper function that simply adds an extra `open: true` tag?

Reply:
I believe this is copied from the other instances of snackbars, which I mostly wrote and/or refactored.

The reason I chose to create this helper function is to encapsulate the behavior of "create / update the current snackbar so that it's like this". The main use case is when the snackbar is shared by a whole deep component tree, so the top-level parent displays it and passes this simple helper down to the deeply nested children. I think it's helpful to have the simple interface (with some extra type safety), so the consumer only has to think about what message and severity the snackbar has, without worrying about implementation details like the `open` state.

Also, the goal is that one day we have a single, site-wide snackbar that can be used by any component (probably through a context-consuming hook), so providing a simple, encapsulated interface should make that adoption path smoother in the future.

Reply:
Just made the necessary changes to give more clarity